src/Substructure.jl

```diff
@@ -0,0 +1,313 @@
+"""
+    has_parents(p, history) -> (boolean, Int64, Int64)
+
+Checks if the jet `p` is a child of two other jets, after clustering 
+
+# Arguments
+- `p`: The jet to check.
+- `history`: The vector of history element obtained from the cluster sequence after clustering.
+
+# Returns
+- (boolean, Int64, Int64): true or false depending on if the jet has a parent or not. If the jet has a parent, returns the indices of the parent jets in the history element. Otherwise, returns -2 (NonexistentParent).
+"""
+function has_parents(p::PseudoJet, history::Vector{HistoryElement})
+    
+    N = p._cluster_hist_index
+    p1 = history[N].parent1
+    p2 = history[N].parent2
+
+    p1 == p2 == NonexistentParent ? result = false : result = true
+    return (result, p1, p2)
+
+end
+
+"""
+    delta_R(jet1, jet2) -> Float64
+
+Function to calculate the distance in the y-ϕ plane between two jets `jet1` and `jet2`
+
+# Arguments
+- `jet1`: The first jet.
+- `jet2`: The second jet.
+
+# Returns
+- Float64: The Euclidean distance in the y-ϕ plane for the two jets.
+"""
+function delta_R(jet1::PseudoJet, jet2::PseudoJet)
```

Comment:
The jet types should not be restricted to `PseudoJet`, they should be `T` where `T <: FourMomentum`,

(This will evolve a bit later, but for now this is correct)

Reply:
Actually, maybe not, if this functionality is only for $pp$ events then it might be better to have the more restricted type...